internal/prober/prober_test.go

```diff
@@ -25,180 +35,631 @@ import (
 	"k8s.io/utils/pointer"
 
 	papi "github.com/gardener/dependency-watchdog/api/prober"
-	mockdiscovery "github.com/gardener/dependency-watchdog/internal/mock/client-go/discovery"
-	mockinterface "github.com/gardener/dependency-watchdog/internal/mock/client-go/kubernetes"
-	mockcoordinationv1 "github.com/gardener/dependency-watchdog/internal/mock/client-go/kubernetes/coordinationv1"
-	mockcorev1 "github.com/gardener/dependency-watchdog/internal/mock/client-go/kubernetes/corev1"
-	mockprober "github.com/gardener/dependency-watchdog/internal/mock/prober"
-	mockscaler "github.com/gardener/dependency-watchdog/internal/mock/prober/scaler"
 )
 
 var (
-	errFoo                   = errors.New("unknown error")
-	proberTestLogger         = logr.Discard()
-	testProbeTimeout         = metav1.Duration{Duration: 3 * time.Millisecond}
-	testProbeInterval        = metav1.Duration{Duration: 3 * time.Millisecond}
-	expiredLeaseRenewTime    = metav1.NewMicroTime(time.Now().Add(-(2 * time.Minute)))
-	nonExpiredLeaseRenewTime = metav1.NewMicroTime(time.Now().Add(-time.Second))
+	testProbeTimeout     = metav1.Duration{Duration: 100 * time.Millisecond}
+	testProbeInterval    = metav1.Duration{Duration: 100 * time.Millisecond}
+	testSeedClientScheme = initializeTestScheme()
 )
 
-type probeTestMocks struct {
-	scaler             *mockscaler.MockScaler
-	shootClientCreator *mockprober.MockShootClientCreator
-	kubernetes         *mockinterface.MockInterface
-	discovery          *mockdiscovery.MockDiscoveryInterface
-	coreV1             *mockcorev1.MockCoreV1Interface
-	coordinationV1     *mockcoordinationv1.MockCoordinationV1Interface
-	node               *mockcorev1.MockNodeInterface
-	lease              *mockcoordinationv1.MockLeaseInterface
-}
-
-type probeTestCase struct {
-	name                    string
-	nodeList                *corev1.NodeList
-	leaseList               *coordinationv1.LeaseList
-	shootClientCreatorError error
-	discoveryError          error
-	nodeListError           error
-	leaseListError          error
-	scaleUpError            error
-	scaleDownError          error
-	minScaleUpCount         int
-	maxScaleUpCount         int
-	minScaleDownCount       int
-	maxScaleDownCount       int
+func initializeTestScheme() *runtime.Scheme {
+	seedClientScheme := *scheme.Scheme
+	_ = v1alpha1.AddToScheme(&seedClientScheme)
+	return &seedClientScheme
 }
 
 func TestAPIServerProbeFailure(t *testing.T) {
-	testCases := []probeTestCase{
-		{name: "Unknown error is returned by api server", discoveryError: errFoo},
```

Comment:
you seems to have removed this test case.

Reply:
It is not needed. The other cases are sufficient.